### PR TITLE
Minion: more robust error handling, and make clippy happy

### PR DESCRIPTION
**Pull request by niklasdewally**
As per #78  - I am removing unwrap and expects from minion_rs, and replacing them with Result<> returning.

### CODE REVIEW

**Review by ozgurakgun — APPROVED** [review]

**Comment by ozgurakgun** [comment]
I tried to resolve conflicts using the web interface, aaand...

Cargo.toml is malformed.

**Comment by ozgurakgun** [comment]
I don't think it's the toml actually, it complains about the lints in the minion toml file

```
Caused by:
  cannot override `workspace.lints` in `lints`, either remove the overrides or `lints.workspace = true` and manually specify the lints
```

**Comment by niklasdewally** [comment]
@ozgurakgun I have rebased locally .

File: solvers/minion/Cargo.toml [hunk 1]
```diff
@@ -6,14 +6,17 @@ edition = "2021"
...
 [build-dependencies]
 cc = { version = "1.0.84", features = ["parallel"] }
 bindgen = "0.69.1"
 glob = "0.3.1"
 
-
-[lints]
-workspace = true
-
+[lints.clippy]
```
Comment (ozgurakgun) [inline solvers/minion/Cargo.toml:17]: I think this is what I got wrong when resolving conflicts on the web.

**Pull request merged**